applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,84 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues);
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
+
+    SetExternalStressVector(rStressVector);
 
     KRATOS_CATCH("")
 }
 
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalDeltaStrainVector(ConstitutiveLaw::Parameters &rValues){
+    const Vector& rStrainVector = rValues.GetStrainVector();
+    std::transform(rStrainVector.begin(), rStrainVector.begin() + mDeltaStrainVector.size(), mStrainVectorFinalized.begin(), mDeltaStrainVector.begin(), std::minus<>{} );
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::UpdateInternalStrainVectorFinalized(ConstitutiveLaw::Parameters &rValues) {
+    this->SetInternalStrainVector(rValues.GetStrainVector());
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetExternalStressVector(Vector& rStressVector){
+    std::copy(mStressVector.begin(), mStressVector.begin() + rStressVector.size(), rStressVector.begin());
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStressVector(const Vector& rStressVector){
+    std::copy(rStressVector.begin(), rStressVector.begin() + mStressVectorFinalized.size(), mStressVectorFinalized.begin());
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::SetInternalStrainVector(const Vector& rStrainVector){
+    std::copy(rStrainVector.begin(), rStrainVector.begin() + mStrainVectorFinalized.size(), mStrainVectorFinalized.begin());
+}
+void GeoLinearElasticPlaneStrain2DLaw::InitializeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+    if (!mIsModelInitialized)
+    {
+        // stress and strain vectors must be initialized:
+        const Vector& rStressVector = rValues.GetStressVector();
+        const Vector& rStrainVector = rValues.GetStrainVector();
+
+        SetInternalStressVector(rStressVector);
+        SetInternalStrainVector(rStrainVector);
+
+        mIsModelInitialized = true;
+    }
+    KRATOS_CATCH("")
+}
+
+void GeoLinearElasticPlaneStrain2DLaw::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters & rValues)
```

Comment:
Could you remove this extra space?
```suggestion
void GeoLinearElasticPlaneStrain2DLaw::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
```

Reply:
done